torch/_dynamo/config.py

```diff
@@ -60,6 +60,10 @@
 # don't specialize on shapes and strides and put shape ops in graph
 dynamic_shapes = os.environ.get("TORCHDYNAMO_DYNAMIC_SHAPES") == "1"
 
+# This flag temporarily changes the behavior of dynamic_shapes=True.
+# When assume_static_by_default is True, we only allocate symbols for shapes marked dynamic via mark_dynamic.
```

Comment:
I think this will die before we move on and forget, but a good thing to do is mention under what condition this knob can be removed